src/Umbraco.Infrastructure/PropertyEditors/MultiUrlPickerValueEditor.cs

```diff
@@ -114,43 +146,38 @@ public IEnumerable<UmbracoEntityReference> GetReferences(object? value)
 
             foreach (LinkDto dto in links)
             {
-                GuidUdi? udi = null;
+                GuidUdi? udi = dto.Udi;
                 var icon = "icon-link";
                 var published = true;
                 var trashed = false;
                 var url = dto.Url;
 
                 if (dto.Udi != null)
                 {
-                    IUmbracoEntity? entity = entities.Find(e => e.Key == dto.Udi.Guid);
-                    if (entity == null)
+                    if (dto.Udi.EntityType == Constants.UdiEntityType.Document)
                     {
-                        continue;
-                    }
+                        url =  _publishedUrlProvider.GetUrl(dto.Udi.Guid, UrlMode.Relative, culture);
+                        IContent? c = _contentService.GetById(dto.Udi.Guid);
```

Comment:
do we have to use content service? 
can't we access nucache for it? If I remember correctly we have published and preview Nucache, so all cases excluding trashed should be covered and just in case it is trashed to hit db?

Reply:
Nucache should not be used by backoffice, we try to make changes that will allow us to have a lazy loaded cache.

Furthermore, as you say nucache also only solves a subset of the requirements

Reply:
Okay  cool, I just think hitting db in this case is not great :C

Reply:
Indeed it is not, but we don't really have all the database available. 
You could argue a lot of this info is not need to have, but it would be very breaking not to return it